monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,324 @@
+        grid_cells = [] 
+        for u0 in range(0, N_COLS):  # 3 columns           
+            x0 =  self.minx + (u0 * cell_size_x)            
+            for v0 in range(0,N_ROWS): # 2 rows
+                y0 = self.miny + (v0 *cell_size_y)
+                x1 = x0 + cell_size_x
+                y1 = y0 + cell_size_y
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))                
+        
+        self.flight_grid = grid_cells
+        
+
+    def generate_flight_paths_points(self):
+
+        ''' For each of the boxes allocated to the operator, get the centroid and buffer to get a flight path. A 70 m radius is provided to have flight paths within each of the boxes '''
+
+        # Iterate over the flight_grid
+        for grid_cell in self.flight_grid:
+            center = grid_cell.centroid
+            ## Transfrom to buffer 140 m diameter circle on which the drone will fly            
+            transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+            transformed_x,transformed_y = transformer.transform(center.x, center.y)
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(70) # build a buffer so that the radius is 70m for the track
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+            for coord in range(0,len(x)):
+                flight_points_with_altitude.append(FlightPoint(lat = x[coord],lng = y[coord],alt= altitude))
+
+            # Build a list of points so that they can be fed to the sim and outputted. 
+            self.flight_points.append(flight_points_with_altitude)
+
+
+class TrackWriter():
+
+    """
+        Write the tracks created by AdjacentCircularFlightPathsGenerator into disk (in the outputs directory) as GeoJSON FeatureCollection 
+        Args:
+        flight_path_points: A set of flight path points generated by generate_flight_paths_points method in the AdjacentCircularFlightPathsGenerator class
+        bboxes: A set of bounding boxes generated by generate_query_bboxes method in the AdjacentCircularFlightPathsGenerator class
+        country_code: An ISO 3166-1 alpha-3 code for a country
+
+        Outputs: 
+        GeoJSON files for bboxes created in the `test_definitions/{country_code}` folder 
+        
+        
+    """
+
+    def __init__(self, path_points, bboxes, country_code='CHE') -> None:
```

Comment:
While this class has a docstring, this method does not -- let's add typing annotations (`def __init__(self, path_points: List[List[FlightPoint]], bboxes: List[QueryBoundingBox], country_code: str = 'CHE')`), or a docstring, or both, since this is the main entrypoint for this class.

Reply:
Done now. thanks!